test/lib/failOnError.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var assert = require('assert');
+var AwsHelper = require('./../../lib/index');
+var simple = require('simple-mock');
+
+describe('AwsHelper.failOnError', function (done) {
+  afterEach(function () {
+    simple.restore();
+  });
+
+  it('should do nothing if there is no error', function (done) {
+    var fakeContext = {
+      fail: function () {}
+    };
+    var spy = simple.spy(fakeContext, 'fail');
+    AwsHelper.failOnError(null, {}, fakeContext);
+    assert(!spy.called);
+    done();
+  });
+
+  it('should call context fail function if error', function (done) {
+    var fakeContext = {
+      fail: function () {}
+    };
+    var fakeError = 'There is an error';
+    var failMock = simple.mock(fakeContext, 'fail');
```

Comment:
:joy: :joy: 
It is. In fact, It was my first idea, and I did it on that way. However then I thought that maybe you would prefer to keep `fail` function as a blackbox so I used the mock. 
But yes, at the end, I ended up with everything a little bit messed. 

I will change it asap.